### PR TITLE
no-inferrable-types: Add ignore-params flag

### PR DESCRIPTION
**Pull request by abierbaum**
This change allows using no-inferrable types at times when you
may want to still require types for method/function parameter lists
(for example when using the typedef rule)

It helps to workaround the issues in #711

### CODE REVIEW

File: src/rules/noInferrableTypesRule.ts [hunk 1]
```diff
@@ -27,13 +27,23 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class NoInferrableTypesWalker extends Lint.RuleWalker {
+    public ignoreParameters: boolean = false;
```
Comment (JKillian) [inline src/rules/noInferrableTypesRule.ts]: Define a `const OPTION_IGNORE_PARAMS = "ignore-params";` above the `class Rule` declaration.

File: src/rules/noInferrableTypesRule.ts [hunk 2]
```diff
@@ -27,13 +27,23 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+
+    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
```
Comment (JKillian) [inline src/rules/noInferrableTypesRule.ts]: And you should be able to delete this constructor all together

File: src/rules/noInferrableTypesRule.ts [hunk 3]
```diff
@@ -27,13 +27,23 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+    }
+
     public visitVariableDeclaration(node: ts.VariableDeclaration) {
         this.checkDeclaration(node);
         super.visitVariableDeclaration(node);
     }
 
     public visitParameterDeclaration(node: ts.ParameterDeclaration) {
-        this.checkDeclaration(node);
+        if (!this.ignoreParameters) {
```
Comment (JKillian) [inline src/rules/noInferrableTypesRule.ts]: then here you can just do `if (!this.hasOption(OPTION_IGNORE_PARAMS))`

**Comment by JKillian** [comment]
Looks good! Can you just make the change noted above to how you handle the option provided? This will make it more consistent with other rules

File: src/rules/noInferrableTypesRule.ts (hunk 1 above)
Comment (abierbaum) [inline src/rules/noInferrableTypesRule.ts]: done.

File: src/rules/noInferrableTypesRule.ts (hunk 2 above)
Comment (abierbaum) [inline src/rules/noInferrableTypesRule.ts]: done

File: src/rules/noInferrableTypesRule.ts (hunk 3 above)
Comment (abierbaum) [inline src/rules/noInferrableTypesRule.ts]: done

**Comment by abierbaum** [comment]
@JKillian I made the changes suggested and pushed up a new commit with everything fixed up.  Let me know if there is anything more to change.

**Comment by JKillian** [comment]
All good, thanks @abierbaum!

**Pull request merged**